File: 2022/tests/day18.test.ts
import { readFileSync } from 'fs';
import { part1, part2 } from '../src/day18';

const dayNumber = __filename.match(/day(\d+)\.test\.ts/)?.[1];
const inputFile: string = `input/day${dayNumber}.txt`;

const example: string = `
2,2,2
1,2,2
3,2,2
2,1,2
2,3,2
2,2,1
2,2,3
2,2,4
2,2,6
1,2,5
3,2,5
2,1,5
2,3,5
`.trim();
const input: string = readFileSync(inputFile, 'utf8');

describe('part 1', () => {
  test('example', () => {
    expect(part1(example)).toBe(64);
  });
  test(`solution`, () => {
    expect(part1(input)).toBe(4482);
  });
});

describe('part 2', () => {
  test('example', () => {
    expect(part2(example)).toBe(58);
  });
  test(`solution`, () => {
    expect(part2(input)).toBe(2576);
  });
});

File: 2024/AdventOfCode/Tests/Day18Test.cs
using NUnit.Framework;

namespace AdventOfCode.Tests;

[TestFixture]
public class Day18Test {
    private const string Example = "Inputs/18-Example.txt";
    private const string Input = "Inputs/18.txt";
    private static Day18 _day;

    [TestFixture]
    public class Part1 {
        [Test]
        public void TestExample() {
            _day = new Day18(Example);
            const string expected = "22";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day18(Input);
            const string expected = "306";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }

    [TestFixture]
    public class Part2 {
        [Test]
        public void TestExample() {
            _day = new Day18(Example);
            const string expected = "6,1";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day18(Input);
            const string expected = "38,63";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }
}